src/extension/exponent/exponentPlatform.ts

```diff
@@ -40,84 +40,86 @@ export class ExponentPlatform extends GeneralMobilePlatform {
         extProps = TelemetryHelper.addPropertyToTelemetryProperties(this.runOptions.reactNativeVersions.reactNativeVersion, "reactNativeVersion", extProps);
         extProps = TelemetryHelper.addPropertyToTelemetryProperties(this.runOptions.expoHostType, "expoHostType", extProps);
 
-        return TelemetryHelper.generate("ExponentPlatform.runApp", extProps, () => {
-            return this.loginToExponentOrSkip(this.runOptions.expoHostType)
-                .then(() =>
-                    XDL.setOptions(this.projectPath, { packagerPort: this.packager.getPort() })
-                )
-                .then(() =>
-                    XDL.startExponentServer(this.projectPath)
-                )
-                .then(() => {
-                    if (this.runOptions.expoHostType !== "tunnel") {
-                        // the purpose of this is to save the same sequence of handling 'adb reverse' command execution as in Expo
-                        // https://github.com/expo/expo-cli/blob/1d515d21200841e181518358fd9dc4c7b24c7cd6/packages/xdl/src/Project.ts#L2226-L2370
-                        // we added this to be sure that our Expo launching logic doesn't have any negative side effects
-                        return XDL.stopAdbReverse(this.projectPath);
-                    }
-                    return XDL.startTunnels(this.projectPath);
-                })
-                .then(() => {
-                    if (this.runOptions.expoHostType !== "local") return false;
-                    // we need to execute 'adb reverse' command to bind ports used by Expo and RN of local machine to ports of a connected Android device or a running emulator
-                    return XDL.startAdbReverse(this.projectPath);
-                })
-                .then((isAdbReversed) => {
-                    switch (this.runOptions.expoHostType) {
-                        case "lan":
-                            return XDL.getUrl(this.projectPath, { dev: true, minify: false, hostType: "lan" });
-                        case "local":
-                            if (isAdbReversed) {
-                                this.logger.info(localize("ExpoStartAdbReverseSuccess", "A device or an emulator was found, 'adb reverse' command successfully executed."));
-                            } else {
-                                this.logger.warning(localize("ExpoStartAdbReverseFailure", "Adb reverse command failed. Couldn't find connected over usb device or running emulator. Also please make sure that there is only one currently connected device or running emulator."));
-                            }
+        return new Promise(() => {
```

Comment:
Why do we use here `Promise.resolve` and `Promise.reject` expressions instead of `new Promise((resolve, reject) =>`?